modules/ROOT/pages/_partials/https-endpoint-authentication-partial.adoc

```diff
@@ -0,0 +1,98 @@
+
+// tag::https-send[]
+// tag::https-receive[]
+[[api-key]]
+= API Key Authentication
+
+Authentication type that requires a valid API key in the HTTP request header. 
+
+Use the following settings to configure API Key authentication:
+
+[%header%autowidth.spread]
+|===
+|Name |Description |Required
+| HTTP Header Name 
+| HTTP header parameter associated with the API key value. 
+| x
+|API Key
+|Value of the API key.
+
+After saving your new or reconfigured endpoint, you cannot subsequently retrieve or modify the API key.
+| x
+|Certificate
+|Certificate to use for authentication.
+|
+|===
+// end::https-send[]
+// https-receive[]
+
+// tag::https-send[]
+// tag:https-receive[]
+[[basic]]
+== Basic Authentication
+
+Authentication type that requires a username and password in the HTTP or HTTP request header.
+
+Use the following settings to configure Basic authentication:
+
+[%header%autowidth.spread]
+|===
+|Name |Description |Required
+|Username
+|Username to use when making a request.
+|x
+|Password
+|Password to use when making a request.
+|x
+|===
+// end::https-send[]]
+// end::https-receive[]
+
+// tag::https-send[]
+[[client-credentials]]
+== Client Credentials Authentication
+
+Authentication type that requires a valid client ID and client secret in the HTTP or HTTPS request header.
+
+Use the following settings to configure Credit Credentials authentication:
+
+[%header%autowidth.spread]
+|===
+|Name |Description |Required
+|Client ID header name
+|HTTP header name that the client ID uses to call the endpoint URL. Default is `"client_id"`.
+|x
+|Client ID header value
+|HTTP header value for the client ID.
+|x
+|Client secret header name
+|HTTP header name for the client secret. Default is `"client_secret"`.
+|x
+|Client secret header value
+|HTTP header value for the client secret.
+|x
+|===
+// end::https-send[]
+
+// tag::https-send[]
+[[oauth-token]]
+== OAuth Token Authentication
+
+Authentication type that requires a valid OAuth token in the HTTP or HTTPS request header. 
+
+Use the following settings to configure OAuth Token authentication:
+
+[%header%autowidth.spread]
+|===
+|Name |Description |Required
+|Token URL
+|URL of the authentication server that provides access tokens.
+|x
+|Client ID 
+|Client ID provided when registering your application with the authorization server.
+|x
+|Client Secret
+|Client secret provided when registering your application with the authorization server.
+|x
+|===
+// end::https-send[]
```

Comment:
General, you are defining the same tag multiple times, that is duplicating the tag, which I think is not intended and could cause issues and build failures. 
The proper way to structure this content would be to define different tags for each section: 
// tag::api-key[]
// tag::basic[]
// tag::client-credentials[]
// tag::oauth-token[]

And then, in each topic that reuses this content, decide which tags to include.

Reply:
The way I use tags is perfectly fine. It worked as intended (even with my accidental omission the second colon in two tags) and I never had a build issue.